# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: docshell/base/CanonicalBrowsingContext.cpp
# Commit: f45996fe15b2
# Full Hash: f45996fe15b23696c88f85a6b154146d3717fc58
# Author: Andrew McCreight <continuation@gmail.com>
# Date: 2021-05-06 03:49:11
# Regressor Bug: 1618547
# File Overlap Count: 1
# Description:
#   Bug 1618547 - Support Fission in the process priority manager. r=kmag,gsvelto
#   
#   With Fission, there can be multiple BrowserParents in a single tab, so
#   this patch moves the tracking of active tabs onto the top BrowsingContext
#   in a tab. If the priority of a top BC is changed, then the activity
# ==============================================================================

diff -r 0e0f90271853 -r f45996fe15b2 docshell/base/CanonicalBrowsingContext.cpp
--- a/docshell/base/CanonicalBrowsingContext.cpp	Wed May 05 20:07:07 2021 +0000
+++ b/docshell/base/CanonicalBrowsingContext.cpp	Wed May 05 20:29:14 2021 +0000
@@ -223,6 +223,11 @@
     BackgroundSessionStorageManager::PropagateManager(Id(), aNewContext->Id());
   }
 
+  // Transfer the ownership of the priority active status from the old context
+  // to the new context.
+  aNewContext->mPriorityActive = mPriorityActive;
+  mPriorityActive = false;
+
   MOZ_ASSERT(aNewContext->mLoadingEntries.IsEmpty());
   mLoadingEntries.SwapElements(aNewContext->mLoadingEntries);
   MOZ_ASSERT(!aNewContext->mActiveEntry);